File: app/src/main/java/com/manimaran/wikiaudio/databases/DBHelper.java
package com.manimaran.wikiaudio.databases;

import android.content.Context;

import androidx.room.Room;

public class DBHelper {

    private static DBHelper mInstance;

    //our app database object
    private AppDatabase appDatabase;

    private DBHelper(Context mCtx) {
        //creating the app database with Room database builder
        //Spell4Wiki-DB is the name of the database
        appDatabase = Room.databaseBuilder(mCtx, AppDatabase.class, "Spell4Wiki-DB").allowMainThreadQueries().build();
    }

    public static synchronized DBHelper getInstance(Context mCtx) {
        if (mInstance == null) {
            mInstance = new DBHelper(mCtx);
        }
        return mInstance;
    }

    public AppDatabase getAppDatabase() {
        return appDatabase;
    }
}

File: app/src/main/java/com/manimarank/spell4wiki/data/db/DBHelper.kt
package com.manimarank.spell4wiki.data.db

import android.content.Context
import androidx.room.Room

class DBHelper private constructor(mCtx: Context) {
    //our app database object
    // creating the app database with Room database builder
    // Spell4Wiki-DB is the name of the database
    val appDatabase: AppDatabase = Room.databaseBuilder(mCtx, AppDatabase::class.java, "Spell4Wiki-DB")
        .allowMainThreadQueries().build()

    companion object {
        private var mInstance: DBHelper? = null
        @Synchronized
        fun getInstance(mCtx: Context): DBHelper {
            if (mInstance == null) {
                mInstance = DBHelper(mCtx)
            }
            return mInstance!!
        }
    }
}